README.md

```diff
@@ -119,7 +119,32 @@ Sometimes it may happen that you need to process the relationship before renderi
 ```ruby
 class BlogPostSerializer < BaseSerializer
   lazy_has_many :comments do |serializer|
-    serializer.lazy_comments.map(&:decorate)
+    -> { serializer.lazy_comments.map(&:decorate) }
+   end
+end
+```
+
+Keep attention that custom finder should be returned in a form of lambda,
+in order to avoid redundant SQL queries when `include_data` AMS setting appears to be `false`:
+
+```ruby
+class BlogPostSerializer < BaseSerializer
+  lazy_has_many :comments do |serializer|
+    include_data :if_sideloaded
+    -> { serializer.lazy_comments.map(&:decorate) }
+   end
+end
+```
+
+Feel free to skip custom lazy finder for association if your goal is just to
+define `include_data` setting nor some links/metas:
+
+```ruby
+class BlogPostSerializer < BaseSerializer
+  lazy_has_many :comments do
+    include_data :if_sideloaded
```

Comment:
Wasn't it supposed to be:

```suggestion
    include_data false
```

?

Reply:
`:if_sideloaded` is the most complex example, this way `data` will be included for explicit `?include=association` and rejected without it.

In second case we need to suppress any kind of finders but we need to keep the finder alive for the time when first case will happen.

i.e.
```ruby
  lazy_has_many :comments do
    include_data :false
    nil
   end
end
```

is completely ok and `ams_lazy_relationship` will behave like that without this PR.

Reply:
Ok, thanks for explanation - so this will work correctly, without N+1s, both when the resource is included and when it isn't 👍